docs/script.md

```diff
@@ -20,41 +16,44 @@ Nextflow scripts have a maximum size of 64 KiB. To avoid this limit for large pi
 
 ## Hello world
 
-To print something is as easy as using one of the `print` or `println` methods.
+You can use the `println` function to print to the console:
 
 ```groovy
-println "Hello, World!"
+println 'Hello, World!'
 ```
 
-The only difference between the two is that the `println` method implicitly appends a newline character to the printed string.
 
 ## Variables
 
-To define a variable, simply assign a value to it:
+Variables are declared using the `def` keyword:
 
 ```groovy
-x = 1
-println x
+def num = 1
+println num
 
-x = new java.util.Date()
-println x
+def date = new java.util.Date()
+println date
 
-x = -3.1499392
+def x = -3.1499392
 println x
 
-x = false
-println x
+def flag = false
+println flag
 
-x = "Hi"
-println x
+def str = "Hi"
+println str
 ```
 
+:::{warning}
+Variables can also be declared without `def` in many cases, but this practice is discouraged outside of simple code snippets because it can lead to a {ref}`race condition <cache-global-var-race-condition>`.
```

Comment:
```suggestion
Variables can also be declared without `def` in many cases. This practice is discouraged outside of simple code snippets as it can lead to a {ref}`race condition <cache-global-var-race-condition>`.
```